.github/workflows/ebpf.yml

```diff
@@ -31,57 +53,86 @@ jobs:
-        working-directory: ${{ github.workspace }}\bpf_performance
-        run: |
-          Release\bpf_performance_runner.exe -i tests.yml -e .sys -r | Tee-Object -FilePath $WorkingDirectory\bpf_performance_native.csv
-
-      - name: Upload BPF performance results
-        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
-        with:
-          name: bpf_performance_native-${{ matrix.os }}-${{ matrix.arch }}
-          path: bpf_performance_native.csv
+    - name: Download ebpf-for-windows
+      uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe
+      with:
+        name: "ebpf-for-windows - MSI installer (Build-x64_NativeOnlyRelease)"
+        path: .
+
+    - name: Install ebpf-for-windows
+      run: |
+        $process = Start-Process -FilePath "msiexec" -ArgumentList "/i ebpf-for-windows.msi /quiet /qn /norestart /log install.log ADDLOCAL=ALL" -Wait -NoNewWindow
+        echo "C:\Program Files\ebpf-for-windows" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+
+    - name: Upload install log
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: install.log
+        path: install.log
+
+    - name: Download bpf_performance repository artifacts
+      run: |
+        Invoke-WebRequest https://github.com/microsoft/bpf_performance/releases/download/v0.0.6/build-Release-windows-2022.zip -OutFile bpf_performance.zip
+
+    - name: Unzip bpf_performance repository artifacts
+      run: |
+        Expand-Archive -Path bpf_performance.zip -DestinationPath .
+
+    - name: Run BPF performance tests
+      run: |
+        if ("${{input.profile}}" -eq "true") {
+          $pre_command = 'wpr.exe -start CPU'
+          $post_command = 'wpr.exe -stop ""${{github.workspace}}\ETL\%NAME%.etl""'
+          mkdir -Force -Path ${{github.workspace}}\ETL
+          Release\bpf_performance_runner.exe -i tests.yml -e .sys -r --pre "$pre_command" --post "$post_command" | Tee-Object -FilePath bpf_performance_native.csv
+        }
+        else {
+          Release\bpf_performance_runner.exe -i tests.yml -e .sys -r | Tee-Object -FilePath bpf_performance_native.csv
+        }
+
+    - name: Upload BPF performance test results
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: bpf_performance_native_${{ matrix.os }}
+        path: bpf_performance_native.csv
+
+    - name: Uninstall ebpf-for-windows
+      if: always()
+      run: |
+        $process = Start-Process -FilePath "msiexec" -ArgumentList "/x ebpf-for-windows.msi /quiet /norestart /log uninstall.log" -Wait -NoNewWindow
+
+    - name: Upload uninstall log
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: uninstall.log
+        path: uninstall.log
+
+    - name: Upload CPU profile
+      if: ${{ inputs.profile == true }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: CPU_Profile
+        path: ETL
```

Comment:
Need to add step to upload results to [bpf_performance dashboard](https://bpfperformancegrafana.azurewebsites.net/d/a40b7e82-a52e-45cc-9477-e43228d10234/ac9e8860-507c-5aa5-8ae5-f092376850a7?orgId=1)

Reply:
Also, should we upload these to the sqlite db?

Reply:
Not quite yet, our plan is to make it 100% mature first for secnetperf before extrapolating to xdp/ebpf. We will likely clean the database as well once we iron out all the small issues. We could port over all the data once its ready :)